domain/secret/state/state.go

```diff
@@ -788,6 +792,9 @@ func updateSecretMetadataFromParams(p domainsecret.UpsertSecretParams, md *secre
 	if p.RotatePolicy != nil {
 		md.RotatePolicyID = int(*p.RotatePolicy)
 	}
+	if p.Checksum != "" {
```

Comment:
We don't need this check - we will always fill out the checksum. 
Migrating older models won't have it, but it will be empty then anyway.

Reply:
Is the commented code above in `pruneUnusedRevisions` able to be removed now?

Reply:
> Is the commented code above in `pruneUnusedRevisions` able to be removed now?

Yes, it can be removed these unused revisions are processed by obsolete flow.